atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -130,6 +151,44 @@ public void close() {
         return getRandomGoodHost().runWithPooledResource(CassandraUtils.getDescribeRing(config));
     }
 
+    private String getSnitch() throws Exception {
+        return getRandomGoodHost().runWithPooledResource(
+                (FunctionCheckedException<CassandraClient, String, Exception>) CassandraClient::describe_snitch);
+    }
+
+    private Set<InetSocketAddress> refreshLocalHosts(List<TokenRange> tokenRanges) {
+        Optional<HostLocation> myLocation = myLocationSupplier.get();
+
+        if (!myLocation.isPresent()) {
+            return ImmutableSet.of();
+        }
+
+        Set<InetSocketAddress> newLocalHosts = Sets.newHashSet();
+        for (TokenRange tokenRange : tokenRanges) {
+            for (EndpointDetails details : tokenRange.getEndpoint_details()) {
+                if (isHostLocal(details, myLocation.get())) {
+                    newLocalHosts.add(getAddressForHostThrowUnchecked(details.getHost()));
+                }
+            }
+        }
+
+        return newLocalHosts;
+    }
+
+    private boolean isHostLocal(EndpointDetails details, HostLocation myLocation) {
```

Comment:
nit: This should be `static`. 

Also, would like to check that `details.isSetHost()` is intentionally included (since it looks like it's not involved in the equals comparison).

Reply:
this check is intentional as I use details.getHost() in the line where it is added to newLocalHosts (so it's not involved in the equals comparison, but is rather an additional check to make sure the endpoint details contains the relevant host).